metis/lib/client/jsx/components/folder-view.jsx

```diff
@@ -35,10 +50,29 @@ const FolderView = ({bucket_name, folder_name}) => {
   const uploadFileInput = useRef(null);
   const uploadDirInput = useRef(null);
 
-  useEffect(() => {
-    invoke({type: 'RETRIEVE_FILES', bucket_name, folder_name});
+  const [loading, setLoading] = useState(true)
+  const [showLoading, setShowLoading] = useState(false)
+
+  const classes = useStyles();
+
+  useAsync( async () => {
+    await invoke({type: 'RETRIEVE_FILES', bucket_name, folder_name});
```

Comment:
Only after properly marking this `invoke` call as an async task did adding latency with my browser's devtools actually let me test that the loading path works.